python/lsst/ts/Dome/mock_controller.py

```diff
@@ -66,17 +68,18 @@ def __init__(
         # Name of a command to report as failed once, the next time it is seen,
         # or None if no failures. Used to test CSC handling of failed commands.
         self.fail_command = None
-        self.status_task = None
-        self.period = 0.1
         # Timeouts used by this class and by its unit test
         self.long_timeout = 20
         self.short_timeout = 2
+        # Time keeping variables
+        self.previous_time_sec = 0
```

Comment:
Please consider this slightly different implementation:
- Specify the time as an argument to determine_statuses, in TAI unix seconds (NOT a duration).
- For mock actuators: record the starting time of each move (and leave it alone until the next move)

Advantages:
- You know at exactly what time status was determined and can provide that time in the status message(s). This is certainly best practice for a real controller.
- It makes unit tests easy to write since you can specify any time you like, and you know the starting time for each actuator so you know what status to expect.

With such a design I don't think it is necessary to record a time here.

Reply:
Done